src/Compilers/CSharp/Portable/SourceGeneration/CSharpSyntaxHelper.cs

```diff
@@ -92,8 +92,16 @@ public override void AddAliases(GreenNode node, ArrayBuilder<(string aliasName,
                 if (global != (usingDirective.GlobalKeyword != null))
                     continue;
 
+                // We only care about aliases from one name to another name.  e.g. `using X = A.B.C;`  That's because
+                // this helper is used to ask "should i bind `[X]`, if the user cares about the metadata type `A.B.C`.
+                // Everything is driven by the user having a fully-qualified-metadata-name to an attribute.  As such,
```

Comment:
I had to read this several times. Consider shortening perhaps to "That's because the caller is only interested in finding a fully-qualified-metadata-name to an attribute."

Reply:
Done.